models/User.js

```diff
@@ -4,82 +4,82 @@ const validator = require("validator");
 
 const toJSON = require("./plugins/toJSON.plugin");
 const userSchema = mongoose.Schema(
-    {
-        username: {
-            type: String,
-            required: true,
-            trim: true,
-        },
-        fullname: {
-            type: String,
-            required: true,
-            trim: true,
-        },
-        phoneNumber: {
-            type: String,
-            trim: true,
-        },
-        email: {
-            type: String,
-            required: true,
-            unique: true,
-            trim: true,
-            lowercase: true,
-            validate(value) {
-                if (!validator.isEmail(value)) {
-                    throw new Error("Invalid email");
-                }
-            },
-        },
-        password: {
-            type: String,
-            required: true,
-            trim: true,
-            validate(value) {
-                if (!value.match(/\d/) || !value.match(/[a-zA-Z]/)) {
-                    throw new Error(
-                        "Password must contain at least one number and one letter"
-                    );
-                }
-            },
-            private: true,
-        },
-        role: {
-            type: String,
-            eum: ["user", "admin"],
-            default: "user",
-        },
-        isEmailVerified: {
-            type: Boolean,
-            default: false,
-        },
-        emailVerificationToken: {
-            type: String,
-        },
+  {
+    username: {
+      type: String,
+      required: true,
+      trim: true,
     },
-    {
-        timestamps: true,
-    }
+    fullname: {
+      type: String,
+      required: true,
+      trim: true,
+    },
+    phoneNumber: {
+      type: String,
+      trim: true,
+    },
+    email: {
+      type: String,
+      required: true,
+      unique: true,
+      trim: true,
+      lowercase: true,
+      validate(value) {
```

Comment:
move these functions to a separate file functions.js or helpers.js
so it can be used in other places also where you need to check valid email

Reply:
Resolved